### ISSUE TITLE
Crash on ObjectBox initialization: Duplicate index ID 1 found for property EntityD.group

### ISSUE THREAD

**Issue by Zpecter** [objectbox/objectbox-swift#1]
### Issue Basics

App crashes when we initialize ObjectBox

- ObjectBox version (are using the latest version?): 0.5.0
- Reproducibility: occurs always

### Reproducing the bug
**Description**

I'm initializing the BoxStore on app start during the initialization of the Swinject graph.

**Code**
```
extension Store {
	static func createStore() throws -> Store {
		let directory = try FileManager.default.url(
			for: .documentDirectory,
			in: .userDomainMask,
			appropriateFor: nil,
			create: true)
		return try Store(directoryPath: directory.path)
	}
}
```

**DBUtils**
```
class DBUtils {
	static var sharedInstance = DBUtils()
	var store : Store!
	
	init() {
		store = try! Store.createStore()
		
		store.register(entity: EntityA.self)
		store.register(entity: EntityB.self)
		store.register(entity: EntityC.self)
		store.register(entity: EntityD.self)
	}
}
```

**Swinject code**
```
container.register(Store.self) { _ in
			DBUtils.sharedInstance.store }.inObjectScope(.container)
```

**Entities**

```
class EntityA: Entity {
	var id : Id<EntityA> = 0
	var name = ""
	var played = 0
	var won = 0
	
	var group: ToOne<EntityC> = nil

	required init() {
	}
}

class EntityB: Entity {
	var id : Id<EntityB> = 0
	var name = ""
	var matches : ToMany<EntityD, EntityB> = nil
	
	required init() {
	}
}

class EntityC: Entity {
	var id : Id<EntityC> = 0
	var name = ""
	var entitiesA: ToMany<EntityA, EntityC> = nil
	
	required init() {
	}
}

class EntityD: Entity {
	var id : Id<EntityD> = 0
    var day : Int?
	var date: Date?
	var isPlayed : Bool = false
	var localPoints: Int?
	var foreignPoints: Int?
	
	var group: ToOne<EntityC> = nil
	var fase: ToOne<EntityB> = nil
	var local: ToOne<EntityA> = nil
	var foreign: ToOne<EntityA> = nil
	
	required init() {
	}
}
```
**Stacktrace**

```
Terminating app due to uncaught exception 'Unhandled ObjectBox C++ Exception', reason: 'Duplicate index ID 1 found for property EntityD.group
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000106cbb1e6 __exceptionPreprocess + 294
	1   libobjc.A.dylib                     0x000000010586b031 objc_exception_throw + 48
	2   ObjectBox                           0x0000000104d13086 _Z24runWithExceptionHandlingPU15__autoreleasingP7NSErrorU13block_pointerFvvE + 2401
	3   ObjectBox                           0x0000000104d1384a _Z24runWithExceptionHandlingPU15__autoreleasingP7NSErrorU13block_pointerFvvE + 4389
	4   MyApp                   0x00000001046c9140 $SSo8OBXStoreC17MyAppE13directoryPath16maxDbSizeInKByte8fileMode0G7ReadersABSS_s6UInt64Vs6UInt32VAKtKcfc + 256
	5   MyApp                   0x00000001046c8fe7 $SSo8OBXStoreC17MyAppE13directoryPath16maxDbSizeInKByte8fileMode0G7ReadersABSS_s6UInt64Vs6UInt32VAKtKcfC + 87
	6   MyApp                   0x00000001046ca76b $SSo8OBXStoreC17MyAppE11createStoreAByKFZ + 411
	7   MyApp                   0x00000001046ca09d $S17MyApp7DBUtilsCACycfc + 173
	8   MyApp                   0x00000001046c9d10 $S17MyApp7DBUtilsCACycfC + 64
	9   MyApp                   0x00000001046c9cbc globalinit_33_9CDCCD402CC08FC843F21FD87EBB68C2_func0 + 28
	10  libdispatch.dylib                   0x000000010b12c848 _dispatch_client_callout + 8
	11  libdispatch.dylib                   0x000000010b12ddc0 dispatch_once_f + 285
	12  libswiftCore.dylib                  0x00000001063feea9 swift_once + 25
	13  MyApp                   0x00000001046c9d94 $S17MyApp7DBUtilsC14sharedInstanceACvau + 36
	14  MyApp                   0x00000001046c1b7f $S17MyApp16ContainerFactoryCACycfcSo8OBXStoreC8Swinject8Resolver_pcfU_ + 15
	15  MyApp                   0x00000001046c1c70 $S8Swinject8Resolver_pSo8OBXStoreCIegno_AaB_pADIegnr_TR + 16
	16  MyApp                   0x00000001046c1cd1 $S8Swinject8Resolver_pSo8OBXStoreCIegno_AaB_pADIegnr_TRTA + 17
	17  Swinject                            0x0000000105084e1f $S8Swinject8Resolver_pxIegnr_AaB_pypIegnr_lTR + 63
	18  Swinject                            0x0000000105084ebd $S8Swinject8Resolver_pxIegnr_AaB_pypIegnr_lTRTA + 29
	19  Swinject                            0x0000000105089428 $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlFypypAA8Resolver_pXEcfU_ + 120
	20  Swinject                            0x000000010508944c $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlFypypAA8Resolver_pXEcfU_TA + 12
	21  Swinject                            0x000000010508668b $Sq_ypIgnr_ypIegyr_q_ypIegnr_ypIeggr_r0_lTR + 75
	22  Swinject                            0x000000010508854d $Sq_ypIgnr_ypIegyr_q_ypIegnr_ypIeggr_r0_lTRTA + 61
	23  Swinject                            0x00000001050866d4 $Sq_ypIegnr_ypIggr_q_ypIegnr_ypIegnr_r0_lTR + 52
	24  Swinject                            0x00000001050885d2 $Sq_ypIegnr_ypIggr_q_ypIegnr_ypIegnr_r0_lTRTA + 50
	25  Swinject                            0x0000000105086c8e $S8Swinject9ContainerC7resolve33_B0248D9BB4D17AC900F06660C57FEB11LL5entry7invokerxSgAA20ServiceEntryProtocol_p_ypq_XEtr0_lF + 1454
	26  Swinject                            0x0000000105085eaf $S8Swinject9ContainerC8_resolve4name6option7invokerxSgSSSg_AA16ServiceKeyOption_pSgypypq_XEctr0_lF + 959
	27  Swinject                            0x0000000105089389 $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlF + 201
	28  Swinject                            0x00000001050892b8 $S8Swinject9ContainerC7resolveyxSgxmlF + 56
	29  Swinject                            0x0000000105089574 $S8Swinject9ContainerCAA8ResolverA2aDP7resolveyqd__Sgqd__mlFTW + 20
	30  SwinjectAutoregistration            0x00000001051dbf44 $S8Swinject8ResolverP0A16AutoregistrationE7resolveqd__SgylF + 68
	31  SwinjectAutoregistration            0x00000001050f88d8 $S8Swinject9ContainerC0A16AutoregistrationE12autoregister_4name11initializerAA12ServiceEntryCyxGxm_SSSgxq_ctr0_lFxAA8Resolver_pcfU_ + 232
	32  SwinjectAutoregistration            0x00000001050f8ad2 $S8Swinject9ContainerC0A16AutoregistrationE12autoregister_4name11initializerAA12ServiceEntryCyxGxm_SSSgxq_ctr0_lFxAA8Resolver_pcfU_TA + 50
	33  Swinject                            0x0000000105084e1f $S8Swinject8Resolver_pxIegnr_AaB_pypIegnr_lTR + 63
	34  Swinject                            0x0000000105084ebd $S8Swinject8Resolver_pxIegnr_AaB_pypIegnr_lTRTA + 29
	35  Swinject                            0x0000000105089428 $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlFypypAA8Resolver_pXEcfU_ + 120
	36  Swinject                            0x000000010508944c $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlFypypAA8Resolver_pXEcfU_TA + 12
	37  Swinject                            0x000000010508668b $Sq_ypIgnr_ypIegyr_q_ypIegnr_ypIeggr_r0_lTR + 75
	38  Swinject                            0x000000010508854d $Sq_ypIgnr_ypIegyr_q_ypIegnr_ypIeggr_r0_lTRTA + 61
	39  Swinject                            0x00000001050866d4 $Sq_ypIegnr_ypIggr_q_ypIegnr_ypIegnr_r0_lTR + 52
	40  Swinject                            0x00000001050885d2 $Sq_ypIegnr_ypIggr_q_ypIegnr_ypIegnr_r0_lTRTA + 50
	41  Swinject                            0x0000000105086c8e $S8Swinject9ContainerC7resolve33_B0248D9BB4D17AC900F06660C57FEB11LL5entry7invokerxSgAA20ServiceEntryProtocol_p_ypq_XEtr0_lF + 1454
	42  Swinject                            0x0000000105085eaf $S8Swinject9ContainerC8_resolve4name6option7invokerxSgSSSg_AA16ServiceKeyOption_pSgypypq_XEctr0_lF + 959
	43  Swinject                            0x0000000105089389 $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlF + 201
	44  Swinject                            0x00000001050892b8 $S8Swinject9ContainerC7resolveyxSgxmlF + 56
	45  Swinject                            0x0000000105089574 $S8Swinject9ContainerCAA8ResolverA2aDP7resolveyqd__Sgqd__mlFTW + 20
	46  SwinjectAutoregistration            0x00000001051dbf44 $S8Swinject8ResolverP0A16AutoregistrationE7resolveqd__SgylF + 68
	47  SwinjectAutoregistration            0x00000001050f9186 $S8Swinject9ContainerC0A16AutoregistrationE12autoregister_4name11initializerAA12ServiceEntryCyxGxm_SSSgxq__q0_t_tctr1_lFxAA8Resolver_pcfU_ + 342
	48  SwinjectAutoregistration            0x00000001050f9535 $S8Swinject9ContainerC0A16AutoregistrationE12autoregister_4name11initializerAA12ServiceEntryCyxGxm_SSSgxq__q0_t_tctr1_lFxAA8Resolver_pcfU_TA + 69
	49  Swinject                            0x0000000105084e1f $S8Swinject8Resolver_pxIegnr_AaB_pypIegnr_lTR + 63
	50  Swinject                            0x0000000105084ebd $S8Swinject8Resolver_pxIegnr_AaB_pypIegnr_lTRTA + 29
	51  Swinject                            0x0000000105089428 $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlFypypAA8Resolver_pXEcfU_ + 120
	52  Swinject                            0x000000010508944c $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlFypypAA8Resolver_pXEcfU_TA + 12
	53  Swinject                            0x000000010508668b $Sq_ypIgnr_ypIegyr_q_ypIegnr_ypIeggr_r0_lTR + 75
	54  Swinject                            0x000000010508854d $Sq_ypIgnr_ypIegyr_q_ypIegnr_ypIeggr_r0_lTRTA + 61
	55  Swinject                            0x00000001050866d4 $Sq_ypIegnr_ypIggr_q_ypIegnr_ypIegnr_r0_lTR + 52
	56  Swinject                            0x00000001050885d2 $Sq_ypIegnr_ypIggr_q_ypIegnr_ypIegnr_r0_lTRTA + 50
	57  Swinject                            0x0000000105086c8e $S8Swinject9ContainerC7resolve33_B0248D9BB4D17AC900F06660C57FEB11LL5entry7invokerxSgAA20ServiceEntryProtocol_p_ypq_XEtr0_lF + 1454
	58  Swinject                            0x0000000105085eaf $S8Swinject9ContainerC8_resolve4name6option7invokerxSgSSSg_AA16ServiceKeyOption_pSgypypq_XEctr0_lF + 959
	59  Swinject                            0x0000000105089389 $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlF + 201
	60  Swinject                            0x00000001050892b8 $S8Swinject9ContainerC7resolveyxSgxmlF + 56
	61  Swinject                            0x0000000105089574 $S8Swinject9ContainerCAA8ResolverA2aDP7resolveyqd__Sgqd__mlFTW + 20
	62  SwinjectAutoregistration            0x00000001051dbf44 $S8Swinject8ResolverP0A16AutoregistrationE7resolveqd__SgylF + 68
	63  SwinjectAutoregistration            0x0000000105100ff0 $S8Swinject9ContainerC0A16AutoregistrationE12autoregister_4name11initializerAA12ServiceEntryCyxGxm_SSSgxq__q0_q1_q2_q3_t_tctr4_lFxAA8Resolver_pcfU_ + 688
	64  SwinjectAutoregistration            0x000000010510192c $S8Swinject9ContainerC0A16AutoregistrationE12autoregister_4name11initializerAA12ServiceEntryCyxGxm_SSSgxq__q0_q1_q2_q3_t_tctr4_lFxAA8Resolver_pcfU_TA + 108
	65  Swinject                            0x0000000105084e1f $S8Swinject8Resolver_pxIegnr_AaB_pypIegnr_lTR + 63
	66  Swinject                            0x0000000105084ebd $S8Swinject8Resolver_pxIegnr_AaB_pypIegnr_lTRTA + 29
	67  Swinject                            0x0000000105089428 $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlFypypAA8Resolver_pXEcfU_ + 120
	68  Swinject                            0x000000010508944c $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlFypypAA8Resolver_pXEcfU_TA + 12
	69  Swinject                            0x000000010508668b $Sq_ypIgnr_ypIegyr_q_ypIegnr_ypIeggr_r0_lTR + 75
	70  Swinject                            0x000000010508854d $Sq_ypIgnr_ypIegyr_q_ypIegnr_ypIeggr_r0_lTRTA + 61
	71  Swinject                            0x00000001050866d4 $Sq_ypIegnr_ypIggr_q_ypIegnr_ypIegnr_r0_lTR + 52
	72  Swinject                            0x00000001050885d2 $Sq_ypIegnr_ypIggr_q_ypIegnr_ypIegnr_r0_lTRTA + 50
	73  Swinject                            0x0000000105086c8e $S8Swinject9ContainerC7resolve33_B0248D9BB4D17AC900F06660C57FEB11LL5entry7invokerxSgAA20ServiceEntryProtocol_p_ypq_XEtr0_lF + 1454
	74  Swinject                            0x0000000105085eaf $S8Swinject9ContainerC8_resolve4name6option7invokerxSgSSSg_AA16ServiceKeyOption_pSgypypq_XEctr0_lF + 959
	75  Swinject                            0x0000000105089389 $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlF + 201
	76  Swinject                            0x00000001050892b8 $S8Swinject9ContainerC7resolveyxSgxmlF + 56
	77  Swinject                            0x0000000105089574 $S8Swinject9ContainerCAA8ResolverA2aDP7resolveyqd__Sgqd__mlFTW + 20
	78  SwinjectAutoregistration            0x00000001051dbf44 $S8Swinject8ResolverP0A16AutoregistrationE7resolveqd__SgylF + 68
	79  SwinjectAutoregistration            0x00000001050fa9d4 $S8Swinject9ContainerC0A16AutoregistrationE12autoregister_4name11initializerAA12ServiceEntryCyxGxm_SSSgxq__q0_q1_t_tctr2_lFxAA8Resolver_pcfU_ + 452
	80  SwinjectAutoregistration            0x00000001050faf43 $S8Swinject9ContainerC0A16AutoregistrationE12autoregister_4name11initializerAA12ServiceEntryCyxGxm_SSSgxq__q0_q1_t_tctr2_lFxAA8Resolver_pcfU_TA + 83
	81  Swinject                            0x0000000105084e1f $S8Swinject8Resolver_pxIegnr_AaB_pypIegnr_lTR + 63
	82  Swinject                            0x0000000105084ebd $S8Swinject8Resolver_pxIegnr_AaB_pypIegnr_lTRTA + 29
	83  Swinject                            0x0000000105089428 $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlFypypAA8Resolver_pXEcfU_ + 120
	84  Swinject                            0x000000010508944c $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlFypypAA8Resolver_pXEcfU_TA + 12
	85  Swinject                            0x000000010508668b $Sq_ypIgnr_ypIegyr_q_ypIegnr_ypIeggr_r0_lTR + 75
	86  Swinject                            0x000000010508854d $Sq_ypIgnr_ypIegyr_q_ypIegnr_ypIeggr_r0_lTRTA + 61
	87  Swinject                            0x00000001050866d4 $Sq_ypIegnr_ypIggr_q_ypIegnr_ypIegnr_r0_lTR + 52
	88  Swinject                            0x00000001050885d2 $Sq_ypIegnr_ypIggr_q_ypIegnr_ypIegnr_r0_lTRTA + 50
	89  Swinject                            0x0000000105086c8e $S8Swinject9ContainerC7resolve33_B0248D9BB4D17AC900F06660C57FEB11LL5entry7invokerxSgAA20ServiceEntryProtocol_p_ypq_XEtr0_lF + 1454
	90  Swinject                            0x0000000105085eaf $S8Swinject9ContainerC8_resolve4name6option7invokerxSgSSSg_AA16ServiceKeyOption_pSgypypq_XEctr0_lF + 959
	91  Swinject                            0x0000000105089389 $S8Swinject9ContainerC7resolve_4namexSgxm_SSSgtlF + 201
	92  Swinject                            0x00000001050892b8 $S8Swinject9ContainerC7resolveyxSgxmlF + 56
	93  MyApp                   0x00000001046a081d $S17MyApp11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0l6LaunchK3KeyaypGSgtFyycfU_ + 189
	94  MyApp                   0x00000001046a0941 $S17MyApp11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0l6LaunchK3KeyaypGSgtFyycfU_TA + 17
	95  MyApp                   0x00000001046a09ad $SIeg_IeyB_TR + 45
	96  libdispatch.dylib                   0x000000010b12c848 _dispatch_client_callout + 8
	97  libdispatch.dylib                   0x000000010b132b35 _dispatch_continuation_pop + 967
	98  libdispatch.dylib                   0x000000010b13dfa6 _dispatch_source_invoke + 1635
	99  libdispatch.dylib                   0x000000010b137b9e _dispatch_main_queue_callback_4CF + 1255
	100 CoreFoundation                      0x0000000106c7dc99 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	101 CoreFoundation                      0x0000000106c41ea6 __CFRunLoopRun + 2342
	102 CoreFoundation                      0x0000000106c4130b CFRunLoopRunSpecific + 635
	103 GraphicsServices                    0x000000010d633a73 GSEventRunModal + 62
	104 UIKit                               0x00000001080f20b7 UIApplicationMain + 159
	105 MyApp                   0x00000001046a2057 main + 71
	106 libdyld.dylib                       0x000000010b1a9955 start + 1
)
```

I also tried without Swinject with the same results. I got the same classes and relationships on Android and it works.

**Comment by greenrobot** [comment]
Impressive that you managed to find something before we officially released. :100: 

We were able to reproduce.

**Comment by greenrobot** [comment]
@Zpecter Please verify, that an update to 0.5.1 fixes the issue. Thank you!

**Comment by Zpecter** [comment]
Yes, now it works. Thank you very much. I have been expecting for a long time to use ObjectBox with Swift, that's why I found the repo before it was released.

**Comment by greenrobot** [comment]
Happy to hear that! Keep us updated how it goes. We still have to go a long way...

**Comment by Zpecter** [comment]
Hi again, I have another problem, some properties on EntityD seems to not be persisted. When I read the objects, day and date are nil, but when I make the put of the entity, these two properties already hava a valid value.

**Comment by uli-objectbox** [comment]
That shouldn't be happening 🤨

How and when are you saving changes to entities? Can you post your code to help us reproduce this issue?

**Comment by Zpecter** [comment]
It's very weird. I have a json file which I read on my AppDelegate to initialize my database. My json file looks like this:

```
[
{
        "PhaseId": 1,
        "Date": "18/09/2018 18:45:00",
        "PointsLocal": null,
        "PointsVisitante": null,
        "GroupId": 1,
        "Day": 1,
        "Played": false,
        "LocalId": 3,
        "ForeignId": 1
    },
...
]
``` 

I read this json file with this function:

```
private func readMatchesJsonFile() -> [Match]? {
	var matches : [Match]? = nil
	if let path = Bundle.main.path(forResource: "Partidos", ofType: "json") {
		do {
			let data = try Data(contentsOf: URL(fileURLWithPath: path), options: .mappedIfSafe)
			let decoder = JSONDecoder()
			decoder.dateDecodingStrategy = .formatted(DateFormatter.iso8601Full)
			matches = try decoder.decode([Match].self, from: data)
		} catch {
			print(error)
		}
	}
	return matches
}

extension DateFormatter {
    static let iso8601Full: DateFormatter = {
        let formatter = DateFormatter()
        formatter.dateFormat = "dd/MM/yyyy HH:mm:ss"
        formatter.calendar = Calendar(identifier: .gregorian)
        formatter.timeZone = TimeZone(secondsFromGMT: 0)
        formatter.locale = Locale(identifier: "en_US_POSIX")
        return formatter
    }()
}

class Match : Decodable {
	var Day : Int? = nil
	var Date: Date? = nil
	var Played : Bool = false
	var PointsLocal: Int? = nil
	var PointsVisitante: Int? = nil
	
	var GroupId: UInt64? = nil
	var PhaseId: UInt64? = nil
	var LocalId: UInt64? = nil
	var ForeignId: UInt64? = nil
	
	required init() {
	}
}
```

When I read the json file, all the fields are filled, and I use these code to iterate over the collection and save the data:

```
for item in matches {
	let match = Match()
	match.day = item.Day
	match.date = item.Date
	match.Played = item.Played
	match.localPoints = item.LocalPoints
	match.foreignPoints = item.ForeignPoints

	if let groupId = match.GroupId, groupId > 0 {
		match.grupo.target = gruposStore?.get(Id<Group>(groupId))
	}
	if let phaseId = item.PhaseId, phaseId > 0 {
		match.phase.target = phasesStore?.get(Id<Phase>(phaseId))
	}
	if let localId = item.LocalId, let foreignId = item.ForeignId, localId > 0, foreignId > 0 {
		match.localTeam.target = teamsStore?.get(Id<Match>(localId))
		match.foreignTeam.target = teamsStore?.get(Id<Match>(foreignId))
	}
	do {
		try matchesStore?.put(match)
	} catch {
	}
}
```

**Comment by Zpecter** [comment]
Another issue that I have is that I can deploy the app to the emulator, but it crashes when I deploy it on my iPhone 6s.

**Comment by uli-objectbox** [comment]
@Zpecter Have you tried updating to 0.5.2? There was a possible issue with device deployment we fixed there.

**Comment by uli-objectbox** [comment]
Also, I assume you have a `print()` in your actual code in the `catch` block and there's no error coming back from `put()`?

**Comment by Zpecter** [comment]
I didn't try version 0.5.2, I'll try later at home, and yes, Match is EntityD.

**Comment by Zpecter** [comment]
I tried the new version, but it still doesn't work. This is the exception it throws:

```
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Tried to register OBXEntityInfo for the same entity name twice.'
```

**Comment by Zpecter** [comment]
Ok, now I can deploy the app to my iphone. I don't see any errors on the ouput, but Day and Date are still nil although when I read the json file, they have a non nil value. Thanks for your help.

**Comment by uli-objectbox** [comment]
OK, I've found part of the issue, there seems to be an issue in our code generator with optionals. I'll look into a fix, in the meantime a workaround would be to make the properties non-optional and use an invalid value to indicate you have no value.

**Comment by uli-objectbox** [comment]
Try replacing the `Pods/ObjectBox/templates/EntityInfo.stencil` file with this one 
[EntityInfo.stencil](https://github.com/objectbox/objectbox-swift/files/2615040/EntityInfo.stencil.txt), that should have the fix for your issue. (You may have to remove a `.txt` extension that Github insists on so it ends in `.stencil` again)

Unless that doesn't work for you, we'll ship that in the next release.

**Comment by Zpecter** [comment]
Hi, I have tested the new file, but it doesn't seem to work. Those properties are still nil. I'll try with non-optional properties.

**Comment by Zpecter** [comment]
I also tried using non-optional properties, but it doesn't work neither.

**Comment by uli-objectbox** [comment]
I just posted 0.5.4 which should fix two more issues that might have affected you. I've also added my test code to the example app. Should this release _not_ work for you, we should probably compare what else is different from the example code in your case, to narrow down what is preventing it from working.

That it wouldn't work with non-optional properties is unusual, though. What (simulated?) device are you building for/running on?
